pkg/stanza/operator/helper/time.go

```diff
@@ -47,12 +47,11 @@ type TimeParser struct {
 
 // Unmarshal starting from default settings
 func (t *TimeParser) Unmarshal(component *confmap.Conf) error {
-	cfg := NewTimeParser()
-	err := component.Unmarshal(&cfg, confmap.WithIgnoreUnused())
+	*t = NewTimeParser()
+	err := component.Unmarshal(t, confmap.WithIgnoreUnused())
```

Comment:
Not sure I understand how do we avoid circular Unmarshal calls here if we send the same data type to  component.Unmarshal `*TimeParser`

Reply:
The way we avoid infinite recursion is by keeping track of which object initiates unmarshaling. If we create and unmarshal a new object every time, then we cannot keep track and we run into a cycle.